gdk-sys/src/lib.rs

```diff
@@ -4047,6 +4038,16 @@ impl ::std::fmt::Debug for GdkWindowRedirect {
     }
 }
 
+#[repr(C)]
+pub struct _GdkAtom(c_void);
+
+impl ::std::fmt::Debug for _GdkAtom {
```

Comment:
This is a bit ugly. If you have a `GdkAtom` (which is a raw pointer), you need to dereference it to use the `Debug` impl. But `GdkAtom` can have values that are invalid to dereference (like 0, 1, 2) IIRC.

Reply:
It shouldn't be really possible to dereference it, because its type is
incomplete. The c_void wrapped in struct doesn't really expresses this now, but
this should be possible to do with extern types once they are in stable Rust.

Reply:
IMHO This is minor problem for `Atom`,
````
let atom = gdk::SELECTION_CLIPBOARD;
println!("{:?}", atom.to_glib_none().0); //GdkAtom
println!("{:?}", atom.name());
```
print
```
0x45
"CLIPBOARD"
```

Reply:
> It shouldn't be really possible to dereference it, because its type is
incomplete.

The `_GdkAtom` type is "incomplete", but `GdkAtom` is a pointer and that can be dereferenced and must be dereferenced to access the `Debug` impl.